src/particles/PhysicsParticles.hpp

```diff
@@ -323,6 +490,71 @@ template <typename ContainerType, ParticleType particleType> class PhysicsPartic
 		}
 	}
 
+	// Implementation of CIC particle creation
+	void createCICParticles(amrex::MultiFab &state, int lev, amrex::Real current_time, amrex::Real dt) const override
+	{
+		if (container_ != nullptr) {
+			const int mass_idx = this->getMassIndex();
+			if (mass_idx >= 0 && mass_idx + 3 < ContainerType::ParticleType::NReal) {
+				CICParticleChecker<problem_t> particle_checker(this->getParam1(), this->getParam2());
```

Comment:
Is the idea that users will then specialize the `CICParticleChecker` for their own problems?